### PR TITLE
WORKSPACE + mam/api: js bindings - integrate and create target

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: mam/api/BUILD [hunk 1]
```diff
@@ -16,6 +17,15 @@ cc_library(
     ],
 )
 
+# Build command:
+# bazel build --spawn_strategy=local --define=workspace=$(bazel info workspace)  --define=output_base=$(bazel info output_base) --crosstool_top='@iota_toolchains//tools/emscripten:toolchain' --cpu=emscripten --host_crosstool_top='@bazel_tools//tools/cpp:toolchain' //mam/api:api.js
```
Comment (oopsmonk) [inline mam/api/BUILD]: put those toolchain configs into `tools/toolchains.rc`

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: mam/api/BUILD (hunk 1 above)
Comment (thibault-martinez) [reply mam/api/BUILD]: What I was going to say!

File: mam/api/BUILD [hunk 2]
```diff
@@ -16,6 +17,15 @@ cc_library(
...
+
+emcc_binary(
```
Comment (thibault-martinez) [inline mam/api/BUILD]: I would see this better in `mam/BUILD` and called `mam.js`, what do you think ?

File: mam/api/BUILD [hunk 3]
```diff
@@ -16,6 +17,15 @@ cc_library(
...
+    name = "api.js",
+    linkopts = ["-s LINKABLE=1 -s EXPORT_ALL=1"],
```
Comment (thibault-martinez) [inline mam/api/BUILD]: Does that mean we are exporting absolutely every function ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/api/BUILD (hunk 3 above)
Comment (tsvisabo) [reply mam/api/BUILD]: yes, every non static function, we can alternatively specify the functions we want exported, no middle solution

File: mam/api/BUILD (hunk 2 above)
Comment (tsvisabo) [reply mam/api/BUILD]: maybe.. anyways, done

**Review by thibault-martinez — DISMISSED** [review]

File: WORKSPACE [hunk 4]
```diff
@@ -44,11 +44,18 @@ load(
...
+
+load("@iota_toolchains//:toolchains.bzl", "setup_initial_deps")
+
+setup_initial_deps()
+
+load("@iota_toolchains//:defs.bzl", "setup_toolchains_repositories")
+
+setup_toolchains_repositories()
+
+load("@iota_toolchains//tools/emscripten:defs.bzl", "emcc_binary")
```
Comment (oopsmonk) [inline WORKSPACE]: remove

File: tools/bazel.rc [hunk 5]
```diff
@@ -42,3 +42,4 @@ build:ubsan --copt -fsanitize=undefined
 build:ubsan --copt -fno-omit-frame-pointer
 build:ubsan --linkopt -fsanitize=undefined
 build:ubsan --linkopt -lubsan
+
```
Comment (oopsmonk) [inline tools/bazel.rc]: remove

**Review by oopsmonk — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**